Source/WTF/wtf/StackTrace.h

```diff
@@ -50,27 +51,42 @@ class PrintStream;
 class StackTrace {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    WTF_EXPORT_PRIVATE static std::unique_ptr<StackTrace> captureStackTrace(int maxFrames, int framesToSkip = 0);
+    WTF_EXPORT_PRIVATE NEVER_INLINE static std::unique_ptr<StackTrace> captureStackTrace(size_t maxFrames, size_t framesToSkip = 0);
 
-    // Borrowed stack trace.
-    StackTrace(void** stack, int size, const char* prefix = "")
-        : m_prefix(prefix)
-        , m_size(size)
-        , m_borrowedStack(stack)
-    { }
+    Span<void* const> stack() const
+    {
+        return Span<void* const> { m_stack + m_initialFrame, m_size };
+    }
+
+    void dump(PrintStream&) const;
+private:
 
-    int size() const { return m_size; }
-    void* const * stack() const
+    StackTrace(size_t size, size_t initialFrame)
+        : m_size(size)
+        , m_initialFrame(initialFrame)
+    {
+    }
+
+    size_t m_size;
+    size_t m_initialFrame;
```

Comment:
You should double check (with a `static_assert`) that `sizeof(size_t) == sizeof(void*)` on `CPU(ARM64_32)`.  Otherwise, use `uintptr_t` here?

Reply:
Checked in the .cpp implementation that uses the fact.